gamla/functional_generic.py

```diff
@@ -392,3 +392,51 @@ async def excepts(*args, **kwargs):
         return excepts
 
     return toolz.excepts(exception, function, handler)
+
+
+find = _compose_over_binary_curried(
+    compose(
+        after(excepts(StopIteration, functional.just(None), toolz.first)),
+        filter,
+    ),
+)
+
+find_index = _compose_over_binary_curried(
+    compose(
+        after(
+            compose_left(
+                functional.star(find),
+                ternary(toolz.identity, toolz.first, functional.just(-1)),
+            ),
+        ),
+        currying.curry(
+            lambda pred, seq: (toolz.compose_left(toolz.second, pred), enumerate(seq)),
+        ),
+    ),
+)
+
+
+take_while = _compose_over_binary_curried(
```

Comment:
iiuc this is non-lazy, is it on purpose?
maybe you can simplify by making a higher order unary function which has lazy behavior